### PR TITLE
Add another metrics middleware which has adjustable sample rate, beca…

### PR DESCRIPTION
**Pull request by johnwu96822**
…use the original metrics middleware would just use the sample rate that the client was configured with

In the case of Shasta where there are many request logs coming in every 30s from every instance. We don't need that many request metrics, maybe only a couple per hour is sufficient for our monitoring purpose.

So add another middleware so the caller can specify sample rate.

- [x] @johnny-lai 
- [x] @supriya

### CODE REVIEW

File: middleware/metrics.go [hunk 1]
```diff
@@ -19,3 +20,23 @@ func Metrics() gin.HandlerFunc {
 		timing.Send()
 	}
 }
+
+//MetricsWithSampleRate is a middleware with adjustable sample rate that will
+//send metrics for every request
+func MetricsWithSampleRate(rate float32) gin.HandlerFunc {
+	if rate < 0 {
+		log.Warnf("Negative metrics sample rate '%f' is invalid. Use the default 1.0", rate)
```
Comment (johnny-lai) [inline middleware/metrics.go]: Since this is a function,  I think it should return `error` if you give it invalid arguments instead of logging.

**Comment by johnny-lai** [comment]
lgtm

**Comment by supriya** [comment]
lgtm

**Comment by johnwu96822** [comment]
[lockit v4.9.2] merged via Lockit CLI on 2019-12-03 14:42:13.747477 -0800 PST m=+6.186410268

**Pull request merged**